### PR TITLE
Support for Ignore Default Values during JSON serialization in System.Text.Json

### PR DESCRIPTION
**Pull request by judeleander**

### CODE REVIEW

**Comment by maryamariyan** [comment]
Thank you for your contribution. As announced in dotnet/coreclr#27549 this repository will be moving to dotnet/runtime on November 13. If you would like to continue working on this PR after this date, the easiest way to move the change to dotnet/runtime is:
1.	In your corefx repository clone, create patch by running `git format-patch origin`
2.	In your runtime repository clone, apply the patch by running `git apply --directory src/corefx <path to the patch created in step 1>`

File: src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleObject.cs [hunk 1]
```diff
@@ -96,7 +97,20 @@ private static bool WriteEndObject(ref WriteStack state)
 
             if (jsonPropertyInfo.ClassType == ClassType.Value)
             {
-                jsonPropertyInfo.Write(ref state, writer);
+                bool includeProperty = true;
```
Comment (layomia) [inline src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleObject.cs]: We need to consider properties of other [class types](https://github.com/dotnet/corefx/blob/cb5be3e9c76c337cb1752b206a50ffc71947aba1/src/System.Text.Json/src/System/Text/Json/Serialization/ClassType.cs) as well.

File: src/System.Text.Json/tests/Serialization/Default.WriteTests.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (layomia) [inline src/System.Text.Json/tests/Serialization/Default.WriteTests.cs]: Nit: this should be named to something like `DefaultValueHandlingTests.cs`.

File: src/System.Text.Json/tests/Serialization/Default.WriteTests.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class DefaultTests
```
Comment (layomia) [inline src/System.Text.Json/tests/Serialization/Default.WriteTests.cs]: Based on API review, we might also need more tests for default value handling, e.g.
- on deserialization
- for collection properties on classes/structs
- for collection elements
- for nested collections

**Review by layomia — COMMENTED** [review]
Thanks for the PR! [API review](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md) is needed for this feature to be implemented in dotnet/runtime.

**Comment by ericstj** [comment]
Thank you for the interest in contributing. This PR is attempting to add public API surface area, and that's something we have very strict processes around, outlined at https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md. We don't accept PRs for new surface area until an issue proposing the new APIs has been discussed, reviewed, and marked api-approved. As such, I'm going to close this PR. Please feel free to open such an issue, following the mentioned process.  Once that’s approved we can accept a PR that implements the new API. Thanks.

**Pull request closed without merging**